src/jobs/closeUnsubmittedRenewalsCheck.ts

```diff
@@ -0,0 +1,161 @@
+ * @param user
+ * @returns BatchJobReport
+ * @example
+ * // returns {
+ *  "jobName":"CLOSING UNSUBMITTED RENEWALS",
+ *  "startedAt":"2023-01-20T08:00:04.817Z",
+ *  "finishedAt":"2023-01-20T08:00:05.394Z",
+ *  "success":true,
+ *  "details":{
+ *    "ids":[],
+ *    "count":0,
+ *    "errors":[],
+ *    "errorCount":0
+ *   }
+ * }
+ */
+async function runCloseUnsubmittedRenewalsCheck(
+  currentDate: Date,
+  user: Identity,
+): Promise<JobReport<BatchJobDetails>> {
+  const jobStartTime = new Date();
+  try {
+    logger.info(`${JOB_NAME} - Initiating...`);
+    const details = await getClosedAppsReportDetails(user, currentDate);
+    details.errors.length
+      ? logger.warn(`${JOB_NAME} - Completed with errors.`)
+      : logger.info(`${JOB_NAME} - Completed.`);
+    const endTime = new Date();
+    const jobSuccessReport: JobReport<BatchJobDetails> = {
+      jobName: JOB_NAME,
+      startedAt: jobStartTime,
+      finishedAt: endTime,
+      success: true,
+      details,
+    };
+    logger.info(`${JOB_NAME} - Report: ${JSON.stringify(jobSuccessReport)}`);
+    return jobSuccessReport;
+  } catch (err) {
+    logger.error(`${JOB_NAME} - Failed to complete, with error: ${(err as Error).message}`);
+    const jobEndTime = new Date();
+    const jobFailedReport: JobReport<BatchJobDetails> = {
+      jobName: JOB_NAME,
+      startedAt: jobStartTime,
+      finishedAt: jobEndTime,
+      success: false,
+      error: `${JOB_NAME} - Failed to complete, with error: ${(err as Error).message}`,
+    };
+    logger.error(`${JOB_NAME} - Report: ${JSON.stringify(jobFailedReport)}`);
+    return jobFailedReport;
+  }
+}
+
+// update app state, including transition to CLOSED + update event
+const closeApplication = async (
+  currentApp: Application,
+  identity: Identity,
+): Promise<Application> => {
+  const appObj = new ApplicationStateManager(currentApp);
+  const role = getDacoRole(identity);
+  logger.info(`${JOB_NAME} - Role ${role} is trying to CLOSE appId ${currentApp.appId}.`);
+  const result = appObj.updateApp({ state: 'CLOSED' }, false, {
+    id: identity.userId,
+    role,
+  });
+  logger.info(`${JOB_NAME} - Updating ${result.appId} in db to ${result.state}.`);
+  const updatedApp = await ApplicationModel.findOneAndUpdate({ appId: result.appId }, result, {
+    new: true,
+  }).exec();
+  if (updatedApp) {
+    return updatedApp;
+  } else {
+    throw new Error(`${JOB_NAME} - Find and update operation failed for ${currentApp.appId}.`);
+  }
+};
+
+export const getAppsClosingQuery = (currentDate: Date): FilterQuery<ApplicationDocument> => {
+  const referenceDate = moment(currentDate).utc().startOf('day');
+  const query: FilterQuery<ApplicationDocument> = {
+    state: {
+      $in: ['DRAFT', 'SIGN AND SUBMIT', 'REVISIONS REQUESTED'],
+    },
+    renewalPeriodEndDateUtc: { $lt: moment(referenceDate).startOf('day').toDate() },
+    isRenewal: true,
+  };
+  return query;
+};
+
+const getClosedAppsReportDetails = async (
```

Comment:
Feel like this is named wrong. Sure, it returns details, but this is actually closing apps!

Reply:
Hmm, i see what you mean, but it's the helper funcs that do the actual state transition...how about something like `closeAppsAndGetReportDetails`?